core/arch/arm/kernel/thread.c

```diff
@@ -253,6 +257,10 @@ void thread_set_exceptions(uint32_t exceptions)
 {
 	uint32_t daif = read_daif();
 
+	/* IRQ must not be unmasked while holding a spinlock */
+	if (!(exceptions & THREAD_EXCP_IRQ))
+		assert_have_no_spinlock();
```

Comment:
i see your point. i tried to find situations where the assert falsely triggs... but you're right. To get a wrong status, something should gone gone wrong earlier. 
Thanks for your answer. (i discard my other related comments).